stacks_queues/queue.py

```diff
@@ -15,47 +18,84 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        # check if full
+        if self.size == self.buffer_size:
+            raise QueueFullException('Queue is full!')
+
+        # check if empty
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+        
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
```

Comment:
👀 It looks like you don't consider that the front might wrap around the buffer it's pointing to `self.size -1`